android/src/main/java/com/media4care/twilio/video/TwilioVideoActivity.java

```diff
@@ -323,9 +327,13 @@ private void createAudioAndVideoTracks() {
         cameraCapturerCompat = new CameraCapturerCompat(this, getAvailableCameraSource());
         localVideoTrack = LocalVideoTrack.create(this, true, cameraCapturerCompat.getVideoCapturer(), LOCAL_VIDEO_TRACK_NAME);
 
+
         audioSwitch.start(
-            (audioDevices, audioDevice) -> {
-                // TODO Enable user select audio device
+            (audioDevices, selectedAudioDevice) -> {
+                if(selectedAudioDevice != null) {
+                    if (selectedAudioDevice instanceof AudioDevice.Earpiece) selectSpeakerAsAudioOutput();
+                    else audioSwitch.selectDevice(selectedAudioDevice);
```

Comment:
is this necessary?

Reply:
The else is not necessary

Reply:
I think so too

Reply:
yes, you are right.  its not needed I thought parameter `selectedAudioDevice` was something that the default active output device